web/packages/teleport/src/DesktopSession/useTdpClientCanvas.tsx

```diff
@@ -89,57 +83,17 @@ export default function useTdpClientCanvas(props: Props) {
     setTdpClient(new TdpClient(addr));
   }, [clusterId, username, desktopName]);
 
-  /**
-   * Synchronize the canvas resolution and display size with the
-   * given ClientScreenSpec.
-   */
-  const syncCanvas = (canvas: HTMLCanvasElement, spec: ClientScreenSpec) => {
-    const { width, height } = spec;
-    canvas.width = width;
-    canvas.height = height;
-    console.debug(`set canvas.width x canvas.height to ${width} x ${height}`);
-    canvas.style.width = `${width}px`;
-    canvas.style.height = `${height}px`;
-    console.debug(
-      `set canvas.style.width x canvas.style.height to ${width} x ${height}`
-    );
-  };
-
-  // Default TdpClientEvent.TDP_PNG_FRAME handler (buffered)
-  const clientOnPngFrame = (
-    ctx: CanvasRenderingContext2D,
-    pngFrame: PngFrame
-  ) => {
-    // The first image fragment we see signals a successful TDP connection.
-    if (!initialTdpConnectionSucceeded.current) {
-      syncCanvas(ctx.canvas, getDisplaySize());
-      setTdpConnection({ status: 'success' });
-      initialTdpConnectionSucceeded.current = true;
-    }
-    ctx.drawImage(pngFrame.data, pngFrame.left, pngFrame.top);
-  };
-
-  // Default TdpClientEvent.TDP_BMP_FRAME handler (buffered)
-  const clientOnBitmapFrame = (
-    ctx: CanvasRenderingContext2D,
-    bmpFrame: BitmapFrame
-  ) => {
-    // The first image fragment we see signals a successful TDP connection.
-    if (!initialTdpConnectionSucceeded.current) {
-      setTdpConnection({ status: 'success' });
-      initialTdpConnectionSucceeded.current = true;
-    }
-    ctx.putImageData(bmpFrame.image_data, bmpFrame.left, bmpFrame.top);
-  };
-
-  // Default TdpClientEvent.TDP_CLIENT_SCREEN_SPEC handler.
-  const clientOnClientScreenSpec = (
-    cli: TdpClient,
-    canvas: HTMLCanvasElement,
-    spec: ClientScreenSpec
-  ) => {
-    syncCanvas(canvas, spec);
-  };
+  const setInitialTdpConnectionSucceeded = useCallback(
+    (callback: () => void) => {
+      // The first image fragment we see signals a successful TDP connection.
+      if (!initialTdpConnectionSucceeded.current) {
+        callback();
+        setTdpConnection({ status: 'success' });
+        initialTdpConnectionSucceeded.current = true;
+      }
+    },
+    [setTdpConnection]
+  );
```

Comment:
i forget why, but we ended up being able to remove any reference or need for an "initial frame" track. at least, i dont have it tracked anymore in avatus/desktop-simplify. but, since i can't remember, we can ignore it. maybe in the future PRs it'll jog my memory again